SQL: panic: (*sem.SubqueryExpr)

## tl;dr

This query causes a panic.

```
$ super -version
Version: 9db7691f8

$ super -c "SELECT COALESCE (1, (MIN (ALL 49)), COUNT (*));"
panic: (*sem.SubqueryExpr) 0xc0004fb050

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic.keySubst({0xa6f1900, 0xc0004fb050}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/projection.go:252 +0xa1a
github.com/brimdata/super/compiler/semantic.keySubst({0xa6f1690, 0xc0004ef440}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/projection.go:180 +0xe3e
github.com/brimdata/super/compiler/semantic.(*translator).genAggregateOutput(0xc0002f4700, {0xa6e68c8, 0xc0004e5c00}, {0xc0004ef140?, 0xc0004ef0c0?, 0xc0004eb8d0?}, {0x0, 0x0, 0x0}, {0xc0004ef1c0, ...})
	/Users/phil/work/super/compiler/semantic/sql.go:232 +0x358
github.com/brimdata/super/compiler/semantic.(*translator).genAggregate(0xc0002f4700, {0x0, 0x2d}, {0xc0004ef140?, 0x0?, 0x1}, {0x0, 0x0}, {0x0, 0x0, ...}, ...)
	/Users/phil/work/super/compiler/semantic/sql.go:206 +0x11d4
github.com/brimdata/super/compiler/semantic.(*translator).sqlSelect(0xc0002f4700, 0xc00042c510, {0x0?, 0x0?, 0xc0004f4510?})
	/Users/phil/work/super/compiler/semantic/sql.go:58 +0x86b
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0x0?, {0xa6efbb0?, 0xc00042c510?}, {0x0?, 0x0?, 0x0?})
	/Users/phil/work/super/compiler/semantic/sql.go:451 +0x28b
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0xc0002f4700, {0xa6efb80?, 0xc0004dee60}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/sql.go:459 +0x1ab
github.com/brimdata/super/compiler/semantic.(*translator).semOp(0xc0002f4700, {0xa6f01e0?, 0xc0004f1d70}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:541 +0xc29
github.com/brimdata/super/compiler/semantic.(*translator).seq(...)
	/Users/phil/work/super/compiler/semantic/op.go:33
github.com/brimdata/super/compiler/semantic.Analyze({0xa6f83e0?, 0xc0004fa4e0?}, 0xc0004f66a0, 0x875caa5?, 0x0)
	/Users/phil/work/super/compiler/semantic/analyzer.go:26 +0x1d0
github.com/brimdata/super/compiler.Analyze(...)
	/Users/phil/work/super/compiler/package.go:27
github.com/brimdata/super/compiler.CompileWithAST(0xc0004fa4e0, 0x888bb65?, 0xc0002f4690, 0x1, 0xc, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/package.go:68 +0x65
github.com/brimdata/super/compiler.(*compiler).NewQuery(0xa6f8488?, 0xc000128320?, 0xc00054f9a8?, {0x0?, 0x90?, 0xa673340?}, 0x87b84a0?)
	/Users/phil/work/super/compiler/compiler.go:42 +0x45
github.com/brimdata/super/runtime.CompileQuery({0xa6f8488?, 0xc000128320?}, 0xc00042c5a0, {0xa6e52a8, 0xc00042a108}, 0xc0004f66a0, {0x0, 0x0, 0x0})
	/Users/phil/work/super/runtime/compiler.go:34 +0x10f
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc000422488, {0xc00003e8f0, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:153 +0x591
github.com/brimdata/super/pkg/charm.path.run({0xc00042a098, 0x1, 0x1}, {0xc00003e8f0, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0xb862380, {0xc00003e8e0, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super commit 9db7691. The query above is a simplification of a query from [a sqllogictest](https://github.com/dolthub/sqllogictest/blob/a1f4aa3c1d691552a4a03d9603c45bcf93d0bb9d/test/random/expr/slt_good_83.test#L40591-L40595).

This query was last seen working at commit 1b72775, right before the merge of #6388.

```
$ super -version
Version: 1b727752f

$ super -c "SELECT COALESCE (1, (MIN (ALL 49)), COUNT (*));"
{COALESCE:1}
```

After #6388 merged, it exhibited the symptoms from #6392, then after the fix for that merged in #6393, now it's got this new panic.

--- a/compiler/semantic/projection.go
+++ b/compiler/semantic/projection.go
@@ -86,6 +86,7 @@ func (a *aggfuncs) subst(e sem.Expr) sem.Expr {
 		return sem.NewThis(e, []string{"in", tmp})
 	case *sem.ArrayExpr:
 		e.Elems = a.substArrayElems(e.Elems)
+	case *sem.BadExpr:
 	case *sem.BinaryExpr:
 		e.LHS = a.subst(e.LHS)
 		e.RHS = a.subst(e.RHS)
@@ -170,6 +171,7 @@ func keySubst(e sem.Expr, exprs []exprloc) (sem.Expr, bool) {
 		panic(e)
 	case *sem.ArrayExpr:
 		e.Elems, ok = keySubstArrayElems(e.Elems, exprs)
+	case *sem.BadExpr:
 	case *sem.BinaryExpr:
 		if e.LHS, ok = keySubst(e.LHS, exprs); ok {
 			e.RHS, ok = keySubst(e.RHS, exprs)
@@ -242,6 +244,7 @@ func keySubst(e sem.Expr, exprs []exprloc) (sem.Expr, bool) {
 		e.From, ok2 = keySubst(e.From, exprs)
 		e.To, ok3 = keySubst(e.To, exprs)
 		ok = ok1 && ok2 && ok3
+	case *sem.SubqueryExpr: // XXX This might need to be traversed?
 	case *sem.ThisExpr:
 		// If we've gotten here it means we have a portion of e that does
 		// not exist in exprs so we are in an error state.

